tutorials/horizontally-scale-mysql-database-backend-with-google-cloud-sql-and-proxysql/index.md

```diff
@@ -0,0 +1,325 @@
+
+1. Declare rules with the following SQL queries:
+
+	```
+	INSERT INTO mysql_rules (rule_id, active, match_digest, destination_hostgroup, apply) VALUES (1,1,'^SELECT.*FOR UPDATE',0,1), (2,1,'^SELECT',1,1);
+	LOAD MYSQL RULES TO RUNTIME;
+	SAVE MYSQL RULES TO DISK;
+	```
+
+	The queries add two rules:
+
+	- Distribute all the `SELECT … FOR UPDATE` queries to database servers in host group `1`
+	- Distribute all the `SELECT` queries to database servers in host group `0`
+
+	Queries that do not match any existing rules go to database servers in host group `0` (default).
+
+2. Add the master instance to host group `1` so that it also helps serve read traffic:
+
+	```
+	INSERT INTO mysql_servers (hostgroup_id, hostname) VALUES (1, '[IP_ADDRESS]');
+	LOAD MYSQL SERVERS TO RUNTIME;
+	SAVE MYSQL SERVERS TO DISK;
+	```
+
+3. Adjust the weights of servers in host group `1`:
+
+	```
+	UPDATE mysql_servers SET weight = 40 WHERE hostname = '[IP_ADDRESS_1]';
+	UPDATE mysql_servers SET weight = 40 WHERE hostname = '[IP_ADDRESS_2]';
+	UPDATE mysql_servers SET weight = 20 WHERE hostname = '[IP_ADDRESS]';
+	LOAD MYSQL SERVERS TO RUNTIME;
+	SAVE MYSQL SERVERS TO DISK;
+	```
+
+	Replace `[IP_ADDRESS_1]`, `[IP_ADDRESS_2]`, `[IP_ADDRESS]` with the ip addresses of read replica #1, read replica #2, and the master instance respectively. The weights are calculated as follows:
+
+	```
+	P(Server # Receives the Query) = (Weight of Server #) / (Total Weight of the Host Group)
+	```
+
+	Read replica #1 has a weight of 40, so its chance of receiving a query is 40/100 = 0.4 (40%). Similarly, read replica #2 has a chance of 40% and the master instance 20%.
+
+### Scaling Dynamically
+
+You can add or remove read replicas any time you want. When you are adding a read replica, create it in Cloud SQL as instructed above, add it to the `mysql_server` table, and update the rules (and/or weights) to route traffic to it. Removing a read replica, however, takes a few additional steps:
+
+1. Update the status of the read replica to `OFFLINE_SOFT`:
+
+	```
+	UPDATE mysql_servers SET status = 'OFFLINE_SOFT' WHERE hostname = '[IP_ADDRESS]';
+	LOAD MYSQL SERVERS TO RUNTIME;
+	SAVE MYSQL SERVERS TO DISK;
+	```
+
+	`OFFLINE_SOFT` status flag instructs ProxySQL to stop sending new queries to the read replica. Existing connections still persist. Monitor the status of the read replica in Cloud Console (Go to the [Cloud SQL Instances page](https://console.cloud.google.com/sql/instances), click the read replica you would like to remove, and the performance chart is available in the Instance Details page). You should see that your read replica is no longer used by your application. Note that setting the read replica to offline in ProxySQL does not stop it from running in Cloud SQL; you are still being charged for the replica. See steps below to remove it from Cloud SQL.
```

Comment:
Take out parenthetical: "Monitor the status of the read replica in Cloud Console by going to the [Cloud SQL Instances page](https://console.cloud.google.com/sql/instances) and clicking the read replica you would like to remove. The performance chart is available in the Instance Details page."